classes/invitation/invitations/BaseInvitation.php

```diff
@@ -252,4 +236,17 @@ public function getId(): int
     {
         return $this->id;
     }
+
+    /**
+     * Get constructor parameters for exclusion.
+     */
+    public function getExcludedUpdatePayloadVariables(): array 
```

Comment:
Would it be cleaner to just keep the invitation subclass (what you're calling a business object) and model separate? We can implement the model code generically, and leave all invitation-specific concerns (both serialized data and functions) to the invitation subclass. Then I think we can let the functions on the subclass to have access to both, and let it get the right data from the right place. I think mixing the data together is going to lead us into confusion.